File: src/main/java/me/ztowne13/customcrates/listeners/CommandPreprocessListener.java
package me.ztowne13.customcrates.listeners;

import me.ztowne13.customcrates.Messages;
import me.ztowne13.customcrates.SpecializedCrates;
import me.ztowne13.customcrates.players.PlayerManager;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;

/**
 * Created by ztowne13 on 3/3/16.
 */
public class CommandPreprocessListener implements Listener {
    SpecializedCrates cc;

    public CommandPreprocessListener(SpecializedCrates cc) {
        this.cc = cc;
    }

    @EventHandler
    public void onCommandPP(PlayerCommandPreprocessEvent e) {
        Player p = e.getPlayer();
        PlayerManager pm = PlayerManager.get(cc, p);

        long ct = System.currentTimeMillis();
        long diff = ct - pm.getCmdCooldown();

        if (diff < 1000 && !pm.getLastCooldown().equalsIgnoreCase("cmd")) {
            e.setCancelled(true);
            Messages.WAIT_ONE_SECOND.msgSpecified(cc, p);
        }

        pm.setLastCooldown("cmd");
        pm.setCmdCooldown(ct);
    }
}

File: src/main/kotlin/me/ztowne13/customcrates/listeners/CommandPreprocessListener.kt
package me.ztowne13.customcrates.listeners

import org.bukkit.event.player.PlayerCommandPreprocessEvent
import me.ztowne13.customcrates.SpecialisedCrates
import me.ztowne13.customcrates.players.PlayerManager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

/**
 * Created by ztowne13 on 3/3/16.
 */
class CommandPreprocessListener(var cc: SpecialisedCrates) : Listener {
    @EventHandler
    fun onCommandPP(e: PlayerCommandPreprocessEvent) {
        val p = e.player
        val pm: PlayerManager = PlayerManager.Companion.get(cc, p)
        val ct = System.currentTimeMillis()
        val diff = ct - pm.cmdCooldown
        if (diff < 1000 && !pm.lastCooldown.equals("cmd", ignoreCase = true)) {
            e.isCancelled = true
            Messages.WAIT_ONE_SECOND.msgSpecified(cc, p)
        }
        pm.lastCooldown = "cmd"
        pm.cmdCooldown = ct
    }
}